lib/Reader.php

```diff
@@ -70,6 +70,12 @@ function parse() : array {
             }
             $result = $this->parseCurrentElement();
 
+            // last line of defense in case errors did occur above
```

Comment:
just to get it right: we check for a error happing in `$this->parseCurrentElement();` with this new case?

Reply:
that's actually true - shall we move it inside of parseCurrentElement()?

Reply:
no we can't as parseCurrentElement() is not expected to throw exceptions.